src/networking/RoleService.ts

```diff
@@ -65,14 +65,27 @@ export default class RoleService extends ModelService {
     ) as Promise<Role[]>;
   }
 
-  async getCurrentRoles(): Promise<Role[]> {
+  async getMostRecentCouncilMemberRoleBySeat(seatId: string): Promise<Role> {
     const populatePersonRef = new Populate(COLLECTION_NAME.Person, REF_PROPERTY_NAME.RolePersonRef);
 
     const networkQueryResponse = this.networkService.getDocuments(
       COLLECTION_NAME.Role,
-      [where("end_datetime", WHERE_OPERATOR.eq, null)],
+      [
+        where(
+          REF_PROPERTY_NAME.RoleSeatRef,
+          WHERE_OPERATOR.eq,
+          doc(NetworkService.getDb(), COLLECTION_NAME.Seat, seatId)
+        ),
+        limit(1),
+        orderBy("end_datetime", ORDER_DIRECTION.desc),
+      ],
       new PopulationOptions([populatePersonRef])
     );
-    return this.createModels(networkQueryResponse, Role, `getCurrentRoles`) as Promise<Role[]>;
+    const roles = await this.createModels(
+      networkQueryResponse,
+      Role,
+      `getMostRecentCouncilMemberRoleBySeat`
+    );
+    return Promise.resolve(roles[0] as Role);
```

Comment:
This could cause a run time error. What if roles.length == 0?

I think it is better to return `roles` for each seat, then when the cards are assembled only create a card for a seat if its associated `roles`.length > 0.

Reply:
The page relies on having matching arrays of seats and roles. I think I will maybe try throwing an error if roles.length == 0, and catching the error in PeoplePage, and there removing the seat-without-a-role from the list of stuff passed down as props?

Reply:
You could still have matching arrays. If `getMostRecentCouncilMemberRoleBySeat` returns an array, then `roles` in PeoplePage would be of type `Role[][]` -- each seat is associated with a Role[] array (which should have only 1 element). If it doesn't don't display the seat?

Reply:
I did it an alternate way: returning Role or null, then stripping the 'bad' Roles from the array and splicing those indices from the seats array.  Either way it's taken care of.